### PR TITLE
Update react/socket

### PR DESCRIPTION
**Pull request by mbonneau**
This PR updates react/socket. There are no API changes.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,10 +11,11 @@
     }
   , "require": {
         "php": ">=5.4"
-      , "react/socket-client": "^0.5"
-      , "ratchet/rfc6455": "^0.2.1"
+      , "react/socket": "^1.0 || ^0.8 || ^0.7"
+      , "evenement/evenement": "^3.0 || ^2.0"
```
Comment (clue) [inline composer.json]: Is this new dependency on Evenement related? I don't see this mentioned in the changeset otherwise.

Comment (mbonneau) [reply composer.json]: This was an existing dependency, when I installed with lowest deps, I found that `WebSocket` was using `EventEmitterTrait`.

Comment (clue) [reply composer.json]: Indeed! Thanks for spotting and pointing this out, no objections in this case :shipit: :-)

**Comment by cboden** [comment]
I ran the Autobahn tests. There were some _major_ slow downs with the newer versions in place. 9.4.1 went from 5 seconds to 15 seconds on my machine. Any ideas @clue?

**Comment by clue** [comment]
@cboden I'm not aware of any noticeable performance regressions in react/socket, except maybe https://github.com/reactphp/socket/pull/114 if you're using an older PHP version? You may be able to lock to older versions and narrow this down to a certain version?

Make sure to let me know if you find anything and I'm happy to address this upstream :+1: That being said, maybe get this in now already and address any performance regressions later? :shipit:

**Comment by cboden** [comment]
I believe I was using 5.4 without TLS. Perhaps the default chunk size changed between versions? I've never been a fan of sacrificing performance for the sake of a release. 😛

**Comment by clue** [comment]
Ouch :stuck_out_tongue: Yes, the above issue affects *all* socket streams for older PHP versions, including plaintext TCP/IP. Can you try `$ composer require react/socket 0.8.2` to lock to an older version and see if the issue goes away?

**Comment by cboden** [comment]
Ah, sorry, I only read the title of the issue. 🔔shame🔔 on me.

I'll test 0.8.2 as well as Autobahn against PHP > 7.1.4 on all 3 iterations to get more information on the performance numbers when I get home from work.

**Comment by cboden** [comment]
I did tests with various versions of PHP and React libs. The 3x slower only happened on a couple tests which I believe is the result of write size buffering. I'm going to conclude that if that much performance is required the developer should be using PHP 7.  :)

**Comment by cboden** [comment]
http://socketo.me/reports/pawl/diff.html

See the 9.x section (Limits/Performance). 

Thoughts?

**Comment by mbonneau** [comment]
Benchmarks on my system for autobahn test 9.4.1 (MacOS 10.13, homebrew php):

Pawl 0.2.3: php 7.1.8, react/stream v0.4.6, react/socket-client v0.5.3: 4560ms
This PR: php 7.1.8, react/stream v0.7.3, react/socket v0.8.4: 4260ms

Pawl 0.2.3: php 7.0.23, react/stream v0.4.6, react/socket-client v0.5.3: 4316ms
This PR: php 7.0.23, react/stream v0.7.3, react/socket v0.8.4: 4271ms

**Pull request merged**